runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkState.java

```diff
@@ -69,6 +68,9 @@ public final class ActiveWorkState {
   @GuardedBy("this")
   private final Map<ShardedKey, Deque<ExecutableWork>> activeWork;
```

Comment:
Seems like we could have a single data structure:
Map<Long /*sharding key*/, LinkedHashMap<WorkId, ExecutableWork>>

That means we don't parallelism sharding key collisions but we also don't expect those (and if it does happen the serialization isn't that big a deal) and it seems like it would make the map operations cheaper to just use the sharding key.

The LinkedHashMap could be used as a Deque since it has insertion ordering and could be used for lookup for failing.

Reply:
done